build: fix package reported by acceptance and compose

tbg opened the pull request:
The acceptance and compose tests were previously reporting `emptypkg` as
their package, which would be used to create issues.

We need to pass the "real" package to go so that it shows up in the
JSON output.

Closes https://github.com/cockroachdb/cockroach/issues/51003.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51005)
<!-- Reviewable:end -->

otan reviewed (approved):
hah, seems a lot cleaner

![nods](https://media2.giphy.com/media/gOkawaguYNiSI/200.gif)

tbg committed:
build: fix package reported by acceptance and compose

The acceptance and compose tests were previously reporting `emptypkg` as
their package, which would be used to create issues.

We need to pass the "real" package to go so that it shows up in the
JSON output.

Closes https://github.com/cockroachdb/cockroach/issues/51003.

Release note: None

tbg commented:
bors r=otan
